wcfsetup/install/files/lib/http/middleware/Xsrf.class.php

```diff
@@ -34,11 +44,45 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
             self::TOKEN_ATTRIBUTE,
             $xsrfToken
         );
+
+        $hasValidXsrfToken = \hash_equals($xsrfToken, $request->getHeaderLine('x-xsrf-token'));
+
         $request = $request->withAttribute(
             self::HAS_VALID_HEADER_ATTRIBUTE,
-            \hash_equals($xsrfToken, $request->getHeaderLine('x-xsrf-token'))
+            $hasValidXsrfToken
         );
 
+        if (
+            $request->getMethod() !== 'GET'
+            && $request->getMethod() !== 'HEAD'
```

Comment:
Perhaps add a helper method `private function isSafeMethod(string $method): bool` that returns true for GET/HEAD and false otherwise.